### PR TITLE
pageserver - configuration file

### PR DESCRIPTION
**Pull request by patins**

### CODE REVIEW

**Review by hlinnaka — COMMENTED** [review]
There are a few references to the environment variables in `test_runner/fixtures/zenith_fixtures.py`, too.

Does this affect the `zenith` binary, or just the `pageserver`? I hope it's just `pageserver`, I think the ZENITH_REPO_DIR env variable still makes sense for the the CLI, it would be tedious to pass the `-D` argument in every call.

Please expand the commit messages with more details of what these commits do and why.

**Comment by patins** [comment]
> There are a few references to the environment variables in `test_runner/fixtures/zenith_fixtures.py`, too.

I think these are only passed to the CLI.

> Does this affect the `zenith` binary, or just the `pageserver`? I hope it's just `pageserver`, I think the ZENITH_REPO_DIR env variable still makes sense for the the CLI, it would be tedious to pass the `-D` argument in every call.

This only changes pageserver behavior, the CLI still uses `POSTGRES_DISTRIB_DIR` and `ZENITH_REPO_DIR`.

> Please expand the commit messages with more details of what these commits do and why.

Will do

File: pageserver/src/bin/pageserver.rs [hunk 1]
```diff
@@ -76,74 +154,59 @@ fn main() -> Result<()> {
...
-    let pg_distrib_dir: PathBuf = {
-        if let Some(postgres_bin) = env::var_os("POSTGRES_DISTRIB_DIR") {
-            postgres_bin.into()
-        } else {
-            let cwd = env::current_dir()?;
-            cwd.join("tmp_install")
-        }
-    };
+    let workdir = Path::new(arg_matches.value_of("workdir").unwrap_or(".zenith"));
+    let cfg_file_path = workdir.canonicalize()?.join("pageserver_config.toml");
```
Comment (kelvich) [inline pageserver/src/bin/pageserver.rs:166]: May be just pageserver.toml or config.toml?

**Review by kelvich — APPROVED** [review]
Looks good, thanks!

**Comment by patins** [comment]
To avoid re-initing, you may need to create `.zenith/pageserver.toml` and populate it with
`pg_distrib_dir = '...'`

**Pull request merged**